### PR TITLE
Add capability to collect "additional" information from hosts

### PR DESCRIPTION
**Pull request by zwass**
Additional information is collected when host details are updated using
the queries specified in the Fleet configuration. This additional
information is then available in the host API responses.

Thank you to [Secureframe](https://www.secureframe.com/) for sponsoring development of this feature.

### CODE REVIEW

File: server/datastore/mysql/app_configs.go [hunk 1]
```diff
@@ -148,8 +149,9 @@ func (d *Datastore) SaveAppConfig(info *kolide.AppConfig) error {
       fim_interval = VALUES(fim_interval),
       fim_file_accesses = VALUES(fim_file_accesses),
       host_expiry_enabled = VALUES(host_expiry_enabled),
-	  host_expiry_window = VALUES(host_expiry_window),
-	  live_query_disabled = VALUES(live_query_disabled)
+   	  host_expiry_window = VALUES(host_expiry_window),
+  	  live_query_disabled = VALUES(live_query_disabled),
+      additional_queries = VALUES(additional_queries)
```
Comment (directionless) [inline server/datastore/mysql/app_configs.go]:
```suggestion
      host_expiry_window = VALUES(host_expiry_window),
      live_query_disabled = VALUES(live_query_disabled),
      additional_queries = VALUES(additional_queries)
```

File: server/datastore/mysql/app_configs.go [hunk 2]
```diff
@@ -117,10 +117,11 @@ func (d *Datastore) SaveAppConfig(info *kolide.AppConfig) error {
       fim_interval,
       fim_file_accesses,
       host_expiry_enabled,
-	  host_expiry_window,
-	  live_query_disabled
+	    host_expiry_window,
+	    live_query_disabled,
+      additional_queries
```
Comment (directionless) [inline server/datastore/mysql/app_configs.go]:
```suggestion
      host_expiry_window,
      live_query_disabled,
      additional_queries
```

**Comment by zwass** [comment]
To mitigate the concerns about perf I did two things:

1) The additional column is not loaded in a call to AuthenticateHost. This means that the hot path (hosts checking in) is less effected by the addition of the column. The column will still be returned through UI/API requests.

2) I added a note to the docs indicating that users should keep the size of data collected small. This, like other scaling concerns for Fleet, will depend on a variety of factors including number of hosts enrolled, intervals, etc.

**Comment by shrav** [comment]
LGTM 🚢

**Pull request merged**